packages/app/prisma-utils/src/plugins/MetricsCollector.ts

```diff
@@ -0,0 +1,170 @@
+import type { PrismaClient } from '@prisma/client'
+import type { Metrics } from '@prisma/client/runtime/library'
+import type { FastifyBaseLogger } from 'fastify'
+import * as prometheus from 'prom-client'
+
+export type PrometheusMetricsDefinitions = {
+  counters: Record<string, prometheus.Counter<'prisma' | 'connection-pool'>>
+  gauges: Record<string, prometheus.Gauge<'prisma' | 'connection-pool'>>
+  histograms: Record<string, prometheus.Histogram<'prisma' | 'connection-pool'>>
+  names: string[]
+}
+
+type PrometheusMetrics = {
+  counters: Record<string, prometheus.MetricObject>
+  gauges: Record<string, prometheus.MetricObject>
+  histograms: Record<string, prometheus.MetricObject>
+}
+
+export type MetricCollectorOptions = {
+  metricsPrefix: string
+}
+
+function registerMetrics(_prefix: string, jsonMetrics: Metrics): PrometheusMetricsDefinitions {
+  const metrics: PrometheusMetricsDefinitions = {
+    counters: {},
+    gauges: {},
+    histograms: {},
+    names: [],
+  }
+  for (const metric of jsonMetrics.counters) {
+    metrics.counters[metric.key] = new prometheus.Counter({
+      name: metric.key,
+      help: metric.description,
+      labelNames: ['prisma', 'connection_pool'] as const,
```

Comment:
prisma has its own labels, we can use them instead + add our prisma label, not all metrics are related to connection pool